Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -45,9 +45,14 @@ namespace AzToolsFramework
         return outputFile.Length() > 0;
     }
 
-    static AZStd::string BookmarkSetRegPath(const AZStd::string& bookmarkFileName, const int index)
+    static AZStd::string BookmarkSetRegPath(const AZStd::string& bookmarkFileName)
     {
-        return "/" + bookmarkFileName + "/LocalBookmarks/" + AZStd::to_string(index);
+        return "/" + bookmarkFileName + "/LocalBookmarks";
+    }
+
+    static AZStd::string BookmarkSetRegPathAtIndex(const AZStd::string& bookmarkFileName, const int index)
+    {
+        return BookmarkSetRegPath(bookmarkFileName) + "/" + AZStd::to_string(index);
```

Comment:
Looking at these now together it seems both functions would benefit from using `AZStd::string::format` to do only 1 operation, instead of several string additions and conversions (which each one result in a new string and its heap allocation).

````
    static AZStd::string BookmarkSetRegPath(const AZStd::string& bookmarkFileName)
    {
        return AZStd::string::format("/%s/LocalBookmarks", bookmarkFileName.c_str());
    }

    static AZStd::string BookmarkSetRegPathAtIndex(const AZStd::string& bookmarkFileName, const int index)
    {
        return AZStd::string::format("%s/%i", BookmarkSetRegPath(bookmarkFileName).c_str(), index);
    }
````

Reply:
Agreed, I was just hesitant to change this right now for fear of breaking something be making a stupid change, let me write a quick test program to convince myself the before/after is the same and I'll update this code before running AR

Reply:
Did a quick test with the code before and after (just because I'm paranoid 😅)

```
    static AZStd::string BookmarkSetRegPath(const AZStd::string& bookmarkFileName)
    {
        return "/" + bookmarkFileName + "/LocalBookmarks";
    }

    static AZStd::string BookmarkSetRegPathAtIndex(const AZStd::string& bookmarkFileName, const int index)
    {
        return BookmarkSetRegPath(bookmarkFileName) + "/" + AZStd::to_string(index);
    }

    static AZStd::string BookmarkSetRegPath2(const AZStd::string& bookmarkFileName)
    {
        return AZStd::string::format("/%s/LocalBookmarks", bookmarkFileName.c_str());
    }

    static AZStd::string BookmarkSetRegPathAtIndex2(const AZStd::string& bookmarkFileName, const int index)
    {
        return AZStd::string::format("%s/%i", BookmarkSetRegPath2(bookmarkFileName).c_str(), index);
    }

const auto str1 = BookmarkSetRegPath("ThisIsAFilename");
const auto str2 = BookmarkSetRegPath2("ThisIsAFilename");
const auto str1a = BookmarkSetRegPathAtIndex("ThisIsAFilename", 1);
const auto str2a = BookmarkSetRegPathAtIndex2("ThisIsAFilename", 1);
```

And all is well 😄 

![image](https://user-images.githubusercontent.com/82228511/159551844-7bf44649-a8c8-4ba5-88b7-1ea0e1cf74c5.png)

Will upload the next diff shortly 👍 then kick off AR

Reply:
Hehe, better to be sure. Well done :)